core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1331,20 +1343,23 @@ public void run() {
             // Otherwise, try to use the default config if available
             if (serviceConfig != null) {
               effectiveServiceConfig = serviceConfig;
-            } else {
+            } else if (defaultServiceConfig != null) {
               effectiveServiceConfig = defaultServiceConfig;
-              if (defaultServiceConfig != null) {
-                channelLogger.log(
-                    ChannelLogLevel.INFO,
-                    "Received no service config, using default service config");
-              }
+              channelLogger.log(
+                  ChannelLogLevel.INFO,
+                  "Received no service config, using default service config");
+            } else {
+              effectiveServiceConfig = EMPTY_SERVICE_CONFIG;
```

Comment:
`onError()` should be put in here. Something like:
```java
if (firstResolutionResponse && configOrError.getError() != null) {
  onError(configOrError.getError());
}
```

Reply:
done.